src/components/Check/index.jsx

```diff
@@ -1,21 +1,31 @@
-import React, { useState } from "react";
+import React from "react";
 import './style.css';
+import { usePrefs } from "../../prefs-context";
 
-const Check = () => {
-  const [checked, setChecked] = useState(false);
-  
+const Check = ({checked, onChange, vegan}) => {
```

Comment:
Teď si řekneme něco o Single Responsibility Princile a obecně správné abstrakci a jak správně designovat komponent/funkce -

Co je účelem komponenty Check? Zobrazit checkbox a umět ho označit (a poslat někam výše změnu toho stavu). K tomu potřebuje jaké údaje(prop)?

checked
onChange
a pokud chceme aby to třeba nešlo vybrat tak třeba
disabled
Ty ale posíláš "vegan", že? Pro obecnou komponentu "Check" je to zbytečně moc informací - to že v aplikaci existují nějaké veganské ingredience je nějaká už tvoje specifická business logika. Takže přemýšlej o tom jako:
Komponenta X umí Y (Check umí být disabled)
a pak té komponentě řekne "kdy" má být disabled

<Check disabled={vegan && veganOnly}
A to na místě, které z logiky věci o tom že existuje nějaký stav Vegan ví.

Vím že je to trochu složitý a maglajz :X ale zkus si to ještě představit tak že jakmile komponentu uděláš moc specifickou tak už jí nepoužiješ jinde v aplikaci, pokud by komponenta CHeck neobsahovala informaci o vega, můžeš jí použít i třeba na masové ingredience, nebo zelenitu atd.

Reply:
Tohle je super, škoda že mě to nenapadlo :) Moc děkuju!